### DISCUSSION TITLE
7 novembre 2018 - esercizio 1

### DISCUSSION THREAD

**Discussion by AglaiaNorza** [sapienzastudentsnetwork/basi-di-dati-1#96] [Esami]
1) Sia data la seguente base di dati relativa ad una compagnia aerea

<div align="center">

AEREO (<ins>ID</ins>, Costruttore, Modello, NPosti, AnnoC, AnnoR) 

VOLO (<ins>Sigla</ins>, Partenza, Arrivo, Orario) 

AEROPORTO (<ins>Sigla</ins>, Città) 

VIAGGIO (<ins>ID</ins>, SiglaVolo, Aereo, Data) 

PERSONALE (<ins>CF</ins>, ID, Nome, Cognome, DataNascita) 

EQUIPAGGIO (<ins>Viaggio, Pers, Data</ins>)

</div>

NOTE

- AEREO.AnnoC e AEREO.AnnoR sono rispettivamente l'anno di costruzione e l'anno di ultima revisione di un aereo
(se l'aereo non è stato mai revisionato AnnoR=00/00/00)

- VOLO.Sigla è la sigla di un volo.

- VOLO.Partenza e VOLO.Arrivo sono sigle di aeroporti.

- Le città più grandi possono avere più aeroporti.

- Un volo viene effettuato sempre alla stessa ora in più date.

- VIAGGIO.SiglaVolo è la sigla di un volo effettuato in data VIAGGIO.Data.

- VIAGGIO.Aereo è l'ID di un aereo.

- EQUIPAGGIO.Viaggio è l'ID di un viaggio.

- EQUIPAGGIO.Pers è l'ID di un membro del personale.

- Un membro del personale può partecipare a più viaggi in date diverse.

1a) Trovare le date del 2018 in cui sono stati effettuati voli diretti a Roma con aerei mai revisionati

1b) Trovare i dati completi di membri del personale che non hanno mai volato a gennaio 2018.

**Comment by marcoseverini** [comment]
![immagine](https://github.com/user-attachments/assets/c119c7d4-ab00-4435-a861-e597664e650b)

**Comment by AglaiaNorza** [comment]
potrebbe avere senso
![es.1](https://github.com/user-attachments/assets/eae5cc86-1ad8-4d51-8c8f-7c7d5ab94083)

**Comment by alem1105** [comment]
1b per farlo abbiamo bisogno della differenza (credo)

![image](https://github.com/user-attachments/assets/13e41a32-12a9-4770-8240-6eb556cd2ebd)

**Comment by Andrii04** [comment]
![IMG_0713](https://github.com/user-attachments/assets/e7eb7421-8870-4268-a61c-45b8bc984753)

non so se è giusto onestamente 👺

**Comment by FeddyLix17** [comment]
1a)

$$\Large Date2018 = \sigma_{VIAGGIO.Data \geq "01/01/2018" \wedge VIAGGIO.Data \leq "31/12/2018"}(VIAGGIO)$$

$$\Large AeroportiRoma = \sigma_{AEROPORTO.Città = "Roma"}(AEROPORTO)$$


$$\Large VARoma= \pi_{VOLO.*}\ VOLO
\underset{
\begin{split} VOLO.Arrivo\\
=  \\ 
AeroportiRoma.Sigla \\ 
\end{split}}{\bowtie}
AeroportiRoma $$

$$\Large VDRoma2018 = Date2018 
\underset{
\begin{split} Date2018.SiglaVolo \\
=  \\ 
VAROMA.Sigla \\ 
\end{split}}{\bowtie}
VARoma$$

$$\Large AereiNRev = \sigma_{AEREO.AnnoR = "00/00/0000"}(AEREO)$$

$$\Large Result = \pi_{VDRoma2018.Data}\ VDRoma2018
\underset{
\begin{split} VDRoma2018.Aereo\\
=  \\ 
AereiNRev .id \\ 
\end{split}}{\bowtie}
AereiNRev$$ 

---

1b)

$$\Large VG2018 = \sigma_{VIAGGIO.Data \geq "01/01/2018" \wedge  VIAGGIO.Data \leq "31/01/2018"}(VIAGGIO)$$

$$\Large EquipVG2018 = EQUIPAGGIO
\underset{
\begin{split} EQUIPAGGIO.Viaggio\\
=  \\ 
VG2018 .ID \\ 
\end{split}}{\bowtie}
VG2018$$

$$\Large PersEquipVG2018 = \pi_{PERSONALE.*} (PERSONALE
\underset{
\begin{split} PERSONALE.ID\\
=  \\ 
EquipVG2018.Pers\\ 
\end{split}}{\bowtie}
EquipVG2018)$$

$$\Large Result = PERSONALE - PersEquipVG2018$$
